include/dali/core/access_order.h

```diff
@@ -103,9 +103,41 @@ class DLL_PUBLIC AccessOrder {
 
   /**
    * @brief Waits in `this` ordering context for the work scheduled in the `other` order.
+   *
+   * Waits for work scheduled in `other` order to complete. If `other` is host, this function
+   * is a no-op.
+   * If either `other` or `this` is a null order, this function has no efect.
+   *
+   * @note `wait` is transitive with a notable exception of null streams, for which the funcion
+   * is a no-op and can break transitivity.
+   *
+   * ```
+   * AccessOrder o1(stream1), o2(stream2), o3(stream3);
+   * o2.wait(o1);
```

Comment:
This only means that all work scheduled on o2 will happen after o1, but when the call completes o1 can be still in progress (like `cudaStreamWaitEvent`), right?

Reply:
Yes, it's exactly like `cudaStreamWaitEvent` because that's the underlying mechanism.